worker/src/main/java/gov/cms/ab2d/worker/service/JobProcessingServiceImpl.java

```diff
@@ -30,28 +60,136 @@ public Job putJobInProgress(String jobId) {
         final Job job = jobRepository.findByJobUuid(jobId);
         Assert.notNull(job, String.format("Job %s not found", jobId));
 
-
         // validate status is SUBMITTED
         if (!SUBMITTED.equals(job.getStatus())) {
-            final String errMsg = String.format("Job %s is not in %s status. Skipping job", jobId, SUBMITTED);
+            final String errMsg = String.format("Job %s is not in %s status.", jobId, SUBMITTED);
             throw new IllegalArgumentException(errMsg);
         }
 
         job.setStatus(IN_PROGRESS);
 
-        log.info("Job [{}] is now IN_PROGRESS", job.getId());
         return jobRepository.save(job);
     }
 
     @Override
-    public void completeJob(Job job) {
+    public Job processJob(final String jobId) {
```

Comment:
Let's add `TransactionPropagation.NEVER` just in case

Reply:
Currently, the method does not have an `@Transactional` annotation.
Would I need to add an `@Transactional` on top of the `processJob()`  to set `TransactionPropagation.NEVER` property?

Reply:
yup

Reply:
done